knowledge-base/treelist-enable-checkbox-selection-only-edit-mode.md

```diff
@@ -0,0 +1,354 @@
+---
+title: Enabling Checkbox Selection Only in Edit Mode for Blazor TreeList
+description: Learn how to show the Checkbox Column in the Telerik Blazor TreeList only during the edit mode and hide it in display mode.
+type: how-to
+page_title: How to Enable Checkbox Selection Only in Edit Mode in Blazor TreeList
+slug: treelist-enable-checkbox-selection-only-edit-mode
+tags: treelist, blazor, checkbox, selection, edit mode, visibility, checkboxcolumn
+res_type: kb
+ticketid: 1657252
+---
+
+## Environment
+
+<table>
+    <tbody>
+        <tr>
+            <td>Product</td>
+            <td>TreeList for Blazor</td>
+        </tr>
+    </tbody>
+</table>
+
+## Description
+
+I want to disable the Checkbox selection in display mode and allow the user to select items only when editing. Is it possible to hide the Checkbox Column in display mode and only show it in the edit mode?
+
+This KB article also answers the following questions:
+- How can I toggle the visibility of the Checkbox Column in the Blazor TreeList?
+- Is it possible to enable Checkbox selection only during edit mode in TreeList?
+- Can I hide the Checkbox Column in display mode for the TreeList?
+
+## Solution
+
+To enable the Checkbox selection only in edit mode in the TreeList component, follow these steps:
+
+1. Enable the [`CheckBoxOnlySelection` parameter]({%slug treelist-columns-checkbox%}#parameters) of the `TreeListCheckboxColumn` to prevent users from selecting rows with a click outside of edit mode.
+2. Bind the [`Visible` parameter]({%slug treelist-columns-checkbox%}#parameters) of the `TreeListCheckboxColumn` to a variable. This allows toggling its visibility based on whether the TreeList is in edit mode.
+3. Handle the CUD (Create, Update, Delete) events to manage the visibility of the `TreeListCheckboxColumn`.
+    - For [`Inline`]({%slug treelist-editing-inline%}) and [`Popup`]({%slug treelist-editing-popup%}) edit modes, this approach works directly. In the [`Incell` edit mode the events flow is a bit different]({%slug treelist-editing-incell%}), so you may need to adjust the logic.
+
+Below is an example implementation that toggles the visibility of the Checkbox Column:
+
+```CSHTML
+@using System.ComponentModel.DataAnnotations @* for the validation attributes *@
```

Comment:
Validation is not relevant to the KB, so we can remove unnecessary stuff to make the example simpler and smaller.